solidity/contracts/Peggy.sol

```diff
@@ -457,17 +454,27 @@ contract Peggy is
         uint256 _amount,
         string calldata _data
     ) external whenNotPaused nonReentrant {
+        uint256 balanceBeforeTransfer = IERC20(_tokenContract).balanceOf(
+            address(this)
+        );
+
         IERC20(_tokenContract).safeTransferFrom(
             msg.sender,
             address(this),
             _amount
         );
+
+        uint256 balanceAfterTransfer = IERC20(_tokenContract).balanceOf(
+            address(this)
+        );
+        uint256 transferAmount = balanceAfterTransfer - balanceBeforeTransfer;
+
```

Comment:
It's not needed because Solidity 0.8 has built-in over-/underflow protection, but yes I did consider adding an additional check, but it's not necessary.